Shared/Tests.Shared/NotificationTests.cs

```diff
@@ -389,6 +389,100 @@ public void List_WhenTransactionHasBothAddAndRemove_ShouldReset()
             Assert.That(propertyEventArgs, Is.EquivalentTo(new[] { "Count", "Item[]" }));
         }
 
+        [TestCase(0, 3, 1, 3, NotifyCollectionChangedAction.Reset)] // a b c d e -> b d a c e
+        [TestCase(0, 3, 0, 2, NotifyCollectionChangedAction.Reset)] // a b c d e -> c d b a e
+        [TestCase(0, 3, 4, 0, NotifyCollectionChangedAction.Reset)] // a b c d e -> e b c d a
+        [TestCase(0, 2, 0, 2, NotifyCollectionChangedAction.Move)] // a b c d e -> c a b d e
+        [TestCase(4, 2, 4, 2, NotifyCollectionChangedAction.Move)] // a b c d e -> a b d e c
+        [TestCase(1, 3, 1, 3, NotifyCollectionChangedAction.Move)] // a b c d e -> c d a b e
+        public void List_WhenTransactionHasMultipleMoves(int oldIndex1, int newIndex1, int oldIndex2, int newIndex2, NotifyCollectionChangedAction expectedAction)
```

Comment:
This is a confusing test to read. I had to stop and think through the test steps a bit too much for me to be comfortable with it as a test clearly describing state and expected result.